coding/csv_file_reader.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <functional>
+#include <string>
+#include <vector>
+
+namespace coding
+{
+class CSVReader
+{
+public:
+  using LineByLineCallback = std::function<void(std::vector<std::string> const & /* line */)>;
+  using FullFileCallback = std::function<void(std::vector<std::vector<std::string>> const & /* file */)>;
+
+  CSVReader(std::string const & dirName = "",
+            bool shouldReadHeader = false) : m_dirName(dirName),
+                                             m_shouldReadHeader(shouldReadHeader) {}
+  
+  void ReadLineByLine(std::string const & fileName, LineByLineCallback const & fn) const;
+  void Read(std::string const & fileName, FullFileCallback const & fn) const;
+  
+  void SetDelimiter(char delimiter) { m_delimiter = delimiter; }
```

Comment:
Please, create a struct named Params. Params should have two fields for now: m_delimiter (space by default) and m_readHeader (true by default). Pass the struct to ReadLineByLine and ReadFull.